test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/BuildIncrementalismTest.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Razor.Design.IntegrationTests
+{
+    public class BuildIncrementalismTest : MSBuildIntegrationTestBase, IClassFixture<BuildServerTestFixture>
+    {
+        private readonly string _pipeName;
+
+        public BuildIncrementalismTest(BuildServerTestFixture fixture)
+        {
+            _pipeName = fixture.PipeName;
+        }
+
+        [Fact]
+        [InitializeTestProject("SimpleMvc")]
+        public async Task BuildIncremental_SimpleMvc_PersistsTagHelperTargetInputFile()
+        {
+            // Build 1
+            var result = await DotnetMSBuild(
+                "Build",
+                $"/p:RazorCompileOnBuild=true /p:UseRazorBuildServer=true /p:_RazorBuildServerPipeName={_pipeName} /p:_RazorForceBuildServer=true",
```

Comment:
Using the server here because we eventually want all these tests to run on the server.

Reply:
Could we instead test the expectation ie. rebuilding twice results in incremental twice? It would cover any other files that affect incrementalism rather than this specific case

Reply:
Yeah I agree with @pranavkm - I don't fundamentally accept that it's a goal that all tests use the server

Reply:
> I don't fundamentally accept that it's a goal that all tests use the server

I misspoke. I didn't mean all the tests. I meant most tests because we'll have build server turned on by default.

Reply:
But this still doesn't need to be hardcoded here.

Reply:
@rynowak I agree this is ugly. I plan to clean this up as part of a separate cleanup. We have other tests that have the hardcoding.